src/main/java/com/uw/config/ApplicationUserService.java

```diff
@@ -0,0 +1,34 @@
+package com.uw.config;
+
+import com.uw.model.User;
+import com.uw.repository.UserRepository;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.transaction.Transactional;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+
+@Service
+public class ApplicationUserService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    public ApplicationUserService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    @Transactional
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user;
+
+        user = userRepository.findByUsername(username)
+                .orElseThrow(EntityNotFoundException::new);
```

Comment:
It's better to throw a custom Exception like UserNotFoundException to be handle properly, providing into the log the user that is being ask. The more info you provide into logs the easier for troubleshooting.